app/src/org/odk/collect/android/widgets/QuestionWidget.java

```diff
@@ -352,7 +355,27 @@ private void stripUnderlines(TextView textView) {
             s.setSpan(span, start, end, 0);
         }
         textView.setText(s);
-    }    
+    }
+
+    public void setQuestionText(TextView textView, FormEntryPrompt prompt){
+        if(prompt.getMarkdownText() != null){
+            textView.setText(forceMarkdown(prompt.getMarkdownText()));
+            textView.setMovementMethod(LinkMovementMethod.getInstance());
+            // Wrap to the size of the parent view
+            textView.setHorizontallyScrolling(false);
+        } else {
+            textView.setText(mPrompt.getLongText());
+        }
+    }
+
+    public void setChoiceText(TextView choiceText, Vector<SelectChoice> choices, int index){
```

Comment:
What about changing the signature to `setChoiceText(TextView choiceText, SelectChoice choice)` and then invoking it like `setChoiceText(label, mItems.get(i)`? 
It seems a bit strange to me to pass in all the choices and the index and deref the choice list inside of the method twice.